includes/admin/class-wc-rest-payments-terminal-locations-controller.php

```diff
@@ -162,7 +162,27 @@ public function get_store_location( $request ) {
 
 			return rest_ensure_response( $this->extract_location_fields( $location ) );
 		} catch ( API_Exception $e ) {
-			return rest_ensure_response( new WP_Error( $e->get_error_code(), $e->getMessage() ) );
+			$error = new WP_Error( $e->get_error_code(), $e->getMessage() );
+			// Stripe will return a 400 for incorrect city, state, or country.
+			// Ideally, we should return a more appropriate error code like 'store_address_is_incorrect',
+			// but that will break older mobile app clients.
+			// Until we have a more granular versioning support for WCPay REST endpoints,
+			// this is the best we can do.
+			if ( 'invalid_request_error' === $e->get_error_code() ) {
```

Comment:
Can we add additional verifications by checking if the current city is empty (see 'get_store_location' for accessing the address). With that, we can differentiate the target case from anything else.

Reply:
Rather `if ( 'invalid_request_error' === $e->get_error_code() && ! isset( $location_address['city'] ) )`.

It should be reliable, but we check different error code here (invalid_request_error). Adding also city check will guarantee it's the issue with the address and not anything else.

Reply:
Ok, I think I got the essence of your suggestion. But `invalid_request_error` is returned by Stripe not only when there are missing address fields but also when a field like city, state or country has a value but one that's incorrect/invalid. IMO, that makes it hard to put additional checks.

For a bit more context, this is what Stripe returns for incorrect address fields:

```json
{
  "error": {
    "message": "Invalid US state or province",
    "param": "address[state]",
    "request_log_url": "https://dashboard.stripe.com/test/logs/req_KEokNlDNgN9317?t=1680801878",
    "type": "invalid_request_error"
  }
}
```

And this is what it returns for missing address fields:

```json
{
  "error": {
    "code": "parameter_missing",
    "doc_url": "https://stripe.com/docs/error-codes/parameter-missing",
    "message": "Missing required address field for a Location in US: address[city].",
    "request_log_url": "https://dashboard.stripe.com/test/logs/req_XgP9IvkGa1wDY0?t=1682398889",
    "type": "invalid_request_error"
  }
}
```

Ideally, the `param` and `code` properties in the above response should be used along with the invalid_request_error check for better reliability. However, `API_Exception` object doesn't forward these fields to us. All it gives us is:

```json
{
    "code": "invalid_request_error",
    "message": "Invalid US state or province",
    "data": null
}
```

So, unless we tweak the `API_Exception` or `Exception` class, these are all the properties we can depend on :(

Or at least that's my understanding.

Reply:
Gotha! Then I'm withdrawing my suggestion for adding more checks here =)